### PR TITLE
Feature/si 2620 check mint with dd attr

### PR DESCRIPTION
**Pull request by LorranSutter**

### CODE REVIEW

**Comment by elffjs** [comment]
@KevinJoiner Good one to look at if you want to see more Solidity.

File: contracts/implementations/MultipleMinter.sol [hunk 1]
```diff
@@ -164,7 +165,6 @@ contract MultipleMinter is
     function mintVehicleAndSdWithDeviceDefinitionSign(
         MintVehicleAndSdWithDdInput calldata data
     ) external onlyRole(MINT_VEHICLE_SD_ROLE) {
-        NodesStorage.Storage storage ns = NodesStorage.getStorage();
```
Comment (elffjs) [inline contracts/implementations/MultipleMinter.sol:167]: Interesting—does this help?

File: contracts/implementations/MultipleMinter.sol [hunk 2]
```diff
@@ -155,6 +155,7 @@ contract MultipleMinter is
      *  manufacturerNode -> Parent manufacturer node id of the vehicle
      *  owner -> The new nodes owner
      *  deviceDefinitionId -> The Device Definition Id
+     *  attrInfoPairsVehicle -> List of attribute-info pairs to be added of the vehicle
```
Comment (elffjs) [inline contracts/implementations/MultipleMinter.sol:158]: Should we also/instead put these docs on the struct?

File: contracts/implementations/nodes/Vehicle.sol [hunk 3]
```diff
@@ -102,17 +102,19 @@ contract Vehicle is AccessControlInternal, VehicleInternal {
     }
 
     /**
-     * @notice Public funtion to mint a vehicle with a Device Definition Id
+     * @notice Function to mint a vehicle with a Device Definition Id
+     * @dev Caller must have the admin role
```
Comment (elffjs) [inline contracts/implementations/nodes/Vehicle.sol:106]: Is that accurate? "Mint vehicle role"?

**Review by elffjs — APPROVED** [review]

File: contracts/implementations/MultipleMinter.sol (hunk 1 above)
Comment (LorranSutter) [reply contracts/implementations/MultipleMinter.sol:167]: Yep... makes the code ugly but it does help to unload the stack.

File: contracts/implementations/MultipleMinter.sol (hunk 2 above)
Comment (LorranSutter) [reply contracts/implementations/MultipleMinter.sol:158]: Good stuff

**Pull request merged**